instrumentation/dropwizard/dropwizard-testing/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizard/DropwizardTest.java

```diff
@@ -0,0 +1,224 @@
+    testSupport.before();
+    return testSupport;
+  }
+
+  @Override
+  protected void stopServer(DropwizardTestSupport<Configuration> server) {
+    server.after();
+  }
+
+  protected void customizeTestOptions(HttpServerTestOptions options) {
+    // this override is needed because dropwizard reports peer ip as the client ip
+    options.setSockPeerAddr(serverEndpoint -> TEST_CLIENT_IP);
+    options.setHasHandlerSpan(endpoint -> endpoint != NOT_FOUND);
+    options.setHasResponseSpan(endpoint -> endpoint == NOT_FOUND);
+    options.setTestPathParam(true);
+    options.setResponseCodeOnNonStandardHttpMethod(405);
+
+    options.setExpectedHttpRoute(
+        (endpoint, method) -> {
+          if (HttpConstants._OTHER.equals(method)) {
+            return getContextPath() + "/*";
+          }
+
+          if (NOT_FOUND.equals(endpoint)) {
+            return getContextPath() + "/*";
+          } else if (PATH_PARAM.equals(endpoint)) {
+            return getContextPath() + "/path/{id}/param";
+          } else {
+            return super.expectedHttpRoute(endpoint, method);
+          }
+        });
+  }
+
+  @Override
+  protected void configure(HttpServerTestOptions options) {
+    super.configure(options);
+    customizeTestOptions(options);
+  }
+
+  Class<? extends Application<Configuration>> testApp() {
+    return TestApp.class;
+  }
+
+  Class<?> testResource() {
+    return ServiceResource.class;
+  }
+
+  @Override
+  protected SpanDataAssert assertHandlerSpan(
+      SpanDataAssert span, String method, ServerEndpoint endpoint) {
+    span.hasName(testResource().getSimpleName() + "." + endpoint.name().toLowerCase(Locale.ROOT))
+        .hasKind(INTERNAL);
+    if (EXCEPTION.equals(endpoint)) {
+      span.hasStatus(StatusData.error()).hasException(new Exception(EXCEPTION.getBody()));
+    }
+    return span;
+  }
+
+  @Override
+  protected SpanDataAssert assertResponseSpan(
+      SpanDataAssert span, SpanData parentSpan, String method, ServerEndpoint endpoint) {
+    span.hasName("CompressedResponseWrapper.sendError").hasKind(INTERNAL);
+    return span;
+  }
+
+  public static class TestApp extends Application<Configuration> {
+    public TestApp() {}
+
+    @Override
+    public void initialize(Bootstrap<Configuration> bootstrap) {}
+
+    @Override
+    public void run(Configuration configuration, Environment environment) {
+      environment.jersey().register(ServiceResource.class);
+    }
+  }
+
+  @Path("/ignored1")
+  interface TestInterface {}
+
+  @Path("/ignored2")
+  public abstract static class AbstractClass implements TestInterface {
+
+    @GET
+    @Path("success")
+    public Response success() {
+      return controller(
+          SUCCESS, () -> Response.status(SUCCESS.getStatus()).entity(SUCCESS.getBody()).build());
+    }
+
+    @GET
+    @Path("query")
+    public Response query_param(@QueryParam("some") String param) {
```

Comment:
A bit surprised that `query_param` doesn't trigger some sort of style warning. It would be more convnetional to call the method `queryParam`.

Reply:
I thought it was strange as well, but when I changed these endpoints to camel case I get an error:

`[Span 1] Expected span to have name <AsyncServiceResource.query_param> but was <AsyncServiceResource.queryParam>`

Which I think is because this is the [name configured to be used](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/3b0e7b8ef2afbccfd6393a03b5a71ed08306888b/testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/ServerEndpoint.java#L36-L37)

I found a [reference in another test](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/3b0e7b8ef2afbccfd6393a03b5a71ed08306888b/instrumentation/spring/spring-webmvc/spring-webmvc-common/testing/src/main/java/io/opentelemetry/instrumentation/spring/webmvc/boot/AbstractSpringBootBasedTest.java#L217-L227) where there is a helper method to reformat and map these particular serverEndpoints back to camel case, so I've implemented something similar.